libs/components/core/src/lib/modules/layout-host/layout-host.directive.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  Directive,
+  ElementRef,
+  RendererFactory2,
+  effect,
+  inject,
+  input,
+} from '@angular/core';
+import { toSignal } from '@angular/core/rxjs-interop';
+
+import { SkyLayoutHostService } from './layout-host.service';
+
+const LAYOUT_FOR_CHILD_CLASS_PREFIX = 'sky-layout-host-for-child-';
+const LAYOUT_CLASS_PREFIX = 'sky-layout-host-';
+
+/**
+ * @internal
+ */
+@Directive({
+  selector: '[skyLayoutHost]',
+  standalone: true,
+  providers: [SkyLayoutHostService],
+})
+export class SkyLayoutHostDirective<T = 'none' | 'fit'> {
+  readonly #elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
+  readonly #layoutForChild = toSignal(
+    inject(SkyLayoutHostService).hostLayoutForChild,
+  );
+  readonly #renderer = inject(RendererFactory2).createRenderer(null, null);
+
+  public readonly layout = input<T>();
+
+  constructor() {
+    effect(() => {
+      const cssClass = [`${LAYOUT_CLASS_PREFIX}${this.layout() ?? 'none'}`];
+      const layoutForChild = this.#layoutForChild()?.layout;
+      if (layoutForChild) {
+        cssClass.push(`${LAYOUT_FOR_CHILD_CLASS_PREFIX}${layoutForChild}`);
+      }
+      this.#elementRef.nativeElement.classList.forEach((className) => {
```

Comment:
Refactor this loop and the one after it to use a `for` loop as it's easier to step through and doesn't have the same overhead of creating/calling a function.

`for (const className of this.#elementRef.nativeElement.classList) {`

Reply:
Good call. Updated.